### PR TITLE
Add erb support

### PR DESCRIPTION
**Pull request by hanachin**
FYI
---

### Plain erb

I tried `erubis -X` and `erb -x` but It doesn't work well in following example.

```erb
<%= content_tag :div do %>
  bar
<% end %>
```

```
% erb -x foo.html.erb
#coding:ASCII-8BIT
_erbout = String.new; _erbout.concat(( content_tag :div do ).to_s); _erbout.concat "\n"
; _erbout.concat "  bar\n"
;  end ; _erbout.concat "\n"
; _erbout.force_encoding(__ENCODING__)
```

```
% erb -x foo.html.erb | ruby -c
-:2: syntax error, unexpected ')'
concat(( content_tag :div do ).to_s); _erbout.concat "\n"
                              ^
-:4: syntax error, unexpected keyword_end, expecting ')'
;  end ; _erbout.concat "\n"
      ^
-:5: syntax error, unexpected end-of-input, expecting ')'
```

### ActionView::Template::Handlers::ERB

It has some noise.

```rb
path = "foo.html.erb"
erb = File.read(path)
template = ActionView::Template.new(erb, path, ActionView::Template::Handlers::ERB.new, {})
puts template.handler.call(template)
# @output_buffer = output_buffer || ActionView::OutputBuffer.new;@output_buffer.append=  content_tag :div do @output_buffer.safe_append='
#   bar
# '.freeze; end
# @output_buffer.to_s
```

### CODE REVIEW

File: lib/querly/pp/cli.rb [hunk 1]
```diff
@@ -70,6 +70,20 @@ def run_haml
...
+        require 'better_html/parser'
+        load_libs
+        source = stdin.read
+        source_buffer = Parser::Source::Buffer.new('(erb)')
+        source_buffer.source = source
+        parser = BetterHtml::Parser.new(source_buffer, template_language: :html)
+        parser.ast.descendants(:erb).each do |erb_node|
+          _, _, code_node, = *erb_node
+          stdout.puts code_node.loc.source
+        end
```
Comment (hanachin) [inline lib/querly/pp/cli.rb]: I referenced code of erb-lint to implement this.
https://github.com/Shopify/erb-lint/blob/04dad0e1a3bac5c68157810e95329929322fc349/lib/erb_lint/processed_source.rb
https://github.com/Shopify/erb-lint/blob/04dad0e1a3bac5c68157810e95329929322fc349/lib/erb_lint/linters/rubocop.rb

**Comment by soutaro** [comment]
@hanachin Thank you very much for opening this pull request. The Rails ERB has been one of the biggest issues on this project since it started, and I really love the idea.

One concern for this patch is the output changes the line numbers of Ruby code. Could you try to fix the patch to preserve the line numbers?

I have an idea to implement naively this by pushing all of the lines to an array and print it later:

```rb
lines = []

parser.ast.descendants(:erb).each do |erb_node|
   _, _, code_node, = *erb_node
  code_node.loc.source.lines.yield_self do |code_lines|
    lines[code_node.loc.line, code_lines.size] = code_lines    # should handle the case when there is more than one code piece in one line.
  end
end

stdout.print lines.join("\n")
```

You said `ActionView::Template::Handlers::ERB` has some noise, but I think it's better than losing line number information even though it contains ERB generated code. (Because one of the first class use cases of this project is to be used in Sider, which requires correct line numbers.)

(I think I can fix the line number problem once it's merged if you don't have time.)

**Comment by hanachin** [comment]
@soutaro Thank you for the feedback! I see the point. I added a 85ba4fd to keep erb code block lineno/column position. That handles the case when there is more than one code piece in one line.

### input example

```
<%= content_tag(
  :div
) do %> <%= "test" %>
  bar
<% end %>
```

### output example

```
    content_tag(
  :div
) do        "test"   
     
   end   
```

**Comment by soutaro** [comment]
@hanachin Greate! I found another edge case:

```erb
<% puts :hello %> <% puts :world %>
```

What do you think about filling `new_source` with `;` instead of ` `?

**Comment by hanachin** [comment]
> What do you think about filling `new_source` with `;` instead of ?

@soutaro Good catch 👍❤️ thank you. I resolved in 46ecb17

**Comment by soutaro** [comment]
Will be shipped in the next release soon. 🎉
